logstash-core/lib/logstash/compiler/lscl.rb

```diff
@@ -478,10 +478,21 @@ class MethodCall < Node; end
 
   class RegexpExpression < Node
     def expr
-      selector, operator_method, regexp = recursive_select(Selector, LogStash::Compiler::LSCL::AST::RegExpOperator, LogStash::Compiler::LSCL::AST::RegExp).map(&:expr)
-
-      raise "Expected a selector #{text_value}!" unless selector
-      raise "Expected a regexp #{text_value}!" unless regexp
+      selector, operator_method, regexp = recursive_select(
+        Selector, 
+        LogStash::Compiler::LSCL::AST::RegExpOperator, 
+        LogStash::Compiler::LSCL::AST::RegExp, 
+        LogStash::Compiler::LSCL::AST::String # Strings work as rvalues! :p
+      ).map(&:expr)
+
+      # Handle string rvalues, they just get turned into regexps
+      # Maybe we really shouldn't handle these anymore...
+      if regexp.class == org.logstash.config.ir.expression.ValueExpression
+        regexp = jdsl.eRegex(regexp.get)
+      end
```

Comment:
I live in a world where I do prefer explicit, I wasn't aware that we were handle theses case.
Can we create an issue that target 6.0?